synapseclient/api/entity_factory.py

```diff
@@ -288,92 +357,43 @@ class type. This will also download the file if `download_file` is set to True.
-                    submission=submission,
-                    synapse_client=synapse_client,
-                )
-            else:  # no filehandle means that we do not have DOWNLOAD permission
-                warning_message = (
-                    "WARNING: You have READ permission on this file entity but not DOWNLOAD "
-                    "permission. The file has NOT been downloaded."
-                )
-                syn.logger.warning(
-                    "\n"
-                    + "!" * len(warning_message)
-                    + "\n"
-                    + warning_message
-                    + "\n"
-                    + "!" * len(warning_message)
-                    + "\n"
-                )
-    elif entity["concreteType"] == concrete_types.TABLE_ENTITY:
-        if not entity_to_update:
-            entity_to_update = Table()
-        entity = entity_to_update.fill_from_dict(
-            entity=entity_bundle["entity"], set_annotations=False
-        )
-    elif entity["concreteType"] == concrete_types.DATASET_ENTITY:
-        if not entity_to_update:
-            from synapseclient.models import Dataset
-
-            entity_to_update = Dataset()
-        entity = entity_to_update.fill_from_dict(
-            entity=entity_bundle["entity"], set_annotations=False
+    # Map concrete types to their corresponding classes
+    ENTITY_TYPE_MAP = {
+        concrete_types.PROJECT_ENTITY: Project,
+        concrete_types.FOLDER_ENTITY: Folder,
+        concrete_types.FILE_ENTITY: File,
+        concrete_types.TABLE_ENTITY: Table,
+        concrete_types.DATASET_ENTITY: Dataset,
+        concrete_types.DATASET_COLLECTION_ENTITY: DatasetCollection,
+        concrete_types.ENTITY_VIEW: EntityView,
+        concrete_types.MATERIALIZED_VIEW: MaterializedView,
+    }
+
+    entity_class = ENTITY_TYPE_MAP.get(entity["concreteType"], None)
+    if not entity_class:
+        raise ValueError(
+            f"Attempting to retrieve an unsupported entity type of {entity['concreteType']}."
         )
-    elif entity["concreteType"] == concrete_types.ENTITY_VIEW:
-        if not entity_to_update:
-            from models.entityview import EntityView
 
-            entity_to_update = EntityView()
-        entity = entity_to_update.fill_from_dict(
-            entity=entity_bundle["entity"], set_annotations=False
+    # Create or use existing entity instance
+    entity_instance = entity_to_update or entity_class()
+
+    # Handle special case for File entities
+    if entity["concreteType"] == concrete_types.FILE_ENTITY:
+        entity_instance = await _handle_file_entity(
+            entity_instance=entity_instance,
+            entity_bundle=entity_bundle,
+            download_file=download_file,
+            download_location=download_location,
+            if_collision=if_collision,
+            submission=submission,
+            synapse_client=syn,
         )
     elif entity["concreteType"] == concrete_types.MATERIALIZED_VIEW:
```

Comment:
That's for making this if/else block simpler. This `MATERIALIZED_VIEW` can be removed now and let the `else` block handle it

Reply:
Thanks, missed this on the merge somehow